src/tanks/Gun.js

```diff
@@ -6,20 +6,38 @@ import {verifyBoolean, verifyDouble} from '../casus/interpreter/Value.js';
 import ImageDrawer from '../battleground/ImageDrawer.js';
 import InterpriterState from '../casus/interpreter/InterpriterState.js';
 import Vec from '../casus/blocks/Vec.js';
+import Bullet from '../battleground/gameobjects/Bullet.js';
 import {
 	TURRET_DIRECTION_VAR_NAME,
 	SHOOT_PRIMARY_WEAPON_VAR_NAME,
+	SHOOT_SECONDARY_WEAPON_VAR_NAME,
 } from '../casus/userInteraction/CasusSpecialVariables.js';
 
 import type Battleground from '../battleground/Battleground.js';
 import type Tank from './Tank.js';
+import type {BulletType} from '../battleground/gameobjects/Bullet.js';
 
 const LERP_PERCENT=0.2;
 const GUN_CENTER_TO_TANK_CENTER=2;
 const GUN_CENTER_TO_GUN_ROT=2;
 const TAU=Math.PI*2;
 const FIRING_SLOWDOWN=0.6;
 
+//
+// Which guns are which types:
```

Comment:
Thank you for this comment. I'm gonna change so the type is recognized in the constructor based on the name instead of having it passed as a parameter.

Reply:
Sounds good to me